src/components/Pages/index.js

```diff
@@ -26,31 +24,36 @@ class VmDetailPage extends React.Component {
     this.requestSent = false
   }
 
-  componentWillMount () {
-    if (Selectors.isFilterChecked()) {
-      this.props.getVms({ vmId: this.props.match.params.id })
+  fetchTheVmIfNeeded () {
+    const { match, vms, getVms } = this.props
+    let requested = false
+
+    if (!this.requestSent && !vms.getIn(['vms', match.params.id])) {
+      this.requestSent = requested = true
+      getVms({ vmId: match.params.id })
     }
+
+    return requested
+  }
+
+  componentWillMount () {
+    this.fetchTheVmIfNeeded()
   }
 
   componentWillUpdate () {
-    const vmInStore = this.props.vms.getIn(['vms', this.props.match.params.id])
-    if (!vmInStore && Selectors.isFilterChecked() && !this.requestSent) {
-      this.requestSent = true
-      this.props.getVms({ vmId: this.props.match.params.id })
-    }
+    this.fetchTheVmIfNeeded()
   }
 
   render () {
-    let { match, vms, config } = this.props
+    const { match, vms, config } = this.props
+
     if (vms.getIn(['vms', match.params.id])) {
       return (<VmDetail vm={vms.getIn(['vms', match.params.id])} config={config} />)
-    } else if (vms.get('loadInProgress')) {
-      console.info(`VmDetailPage: VM id cannot be found: ${match.params.id}. Load is still in progress - waiting before redirect`)
-      return null
     }
 
-    console.info(`VmDetailPage: VM id cannot be found: ${match.params.id}. Redirecting to / `)
-    return <Redirect to='/' />
+    // TODO: Add handling for if the fetch runs but fails (FETCH-FAIL)
```

Comment:
Yep, let's not forget this.

Reply:
I'll address this in a followup PR.  I will create an issue for it later today.  It'll be something like "Handle 404 conditions gracefully".

Reply:
#631

Reply:
#631

Reply:
ok